cmake/kokkos_cxx.cmake

```diff
@@ -211,6 +188,37 @@ ELSE()
   ENDIF()
 ENDIF()
 
+# Enforce that extensions are turned off for nvcc_wrapper.
+# For compiling CUDA code using nvcc_wrapper, we will use the host compiler's
+# flags for turning on C++11.  Since for compiler ID and versioning purposes
+# CMake recognizes the host compiler when calling nvcc_wrapper, this just
+# works.  Both NVCC and nvcc_wrapper only recognize '-std=c++11' which means
+# that we can only use host compilers for CUDA builds that use those flags.
+# It also means that extensions (gnu++11) can't be turned on for CUDA builds.
+IF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
```

Comment:
Any reason why that block wouldn't be within the `if(KOKKOS_ENABLE_CUDA)` block below?

Reply:
In theory we could enable compilation with nvcc without CUDA being on (though it doesn't work right now) and NVCC doesn't know what gnu++11 is.

Reply:
Is that something we want to support?

Reply:
This decision is above my pay grade.  I would personally throw either 1) throw an error in CUDA is not enable but nvcc is the compiler or 2) warn and enable CUDA.